src/containers/find-offer/offer-request-block/OfferRequestBlock.tsx

```diff
@@ -1,30 +1,41 @@
 import { useTranslation } from 'react-i18next'
+import { useAppSelector } from '~/hooks/use-redux'
+import { UserRoleEnum } from '~/types'
+import { useDrawer } from '~/hooks/use-drawer'
+import useBreakpoints from '~/hooks/use-breakpoints'
 
+import AppButton from '~/components/app-button/AppButton'
+import CreateOfferRequest from '../create-offer-request/CreateOfferRequest'
 import TitleBlock from '~/components/title-block/TitleBlock'
 import icon from '~/assets/img/find-offer/subject_icon.png'
-import AppButton from '~/components/app-button/AppButton'
-import useBreakpoints from '~/hooks/use-breakpoints'
-import { useDrawer } from '~/hooks/use-drawer'
-import { translationKey } from '~/containers/find-offer/constants'
+import AppDrawer from '~/components/app-drawer/AppDrawer'
+
+const translationKey = 'findOffers.offerRequestBlock'
 
 const OfferRequestBlock = () => {
   const { t } = useTranslation()
   const { isMobile } = useBreakpoints()
-  const { openDrawer } = useDrawer()
+  const { openDrawer, closeDrawer, isOpen } = useDrawer()
 
-  const handleOpenDrawer = () => {
-    openDrawer()
-  }
+  const userRole = useAppSelector((state) => state.appMain.userRole)
+  const buttonText =
+    userRole === UserRoleEnum.Student
+      ? `${translationKey}.button.student`
+      : `${translationKey}.button.tutor`
 
   return (
     <TitleBlock img={icon} translationKey={translationKey}>
       <AppButton
         fullWidth={isMobile}
-        onClick={handleOpenDrawer}
+        onClick={() => openDrawer()}
```

Comment:
just onClick={openDrawer} would be better

Reply:
The problem is that the `onClick` handler for the AppButton expects a function that takes a MouseEvent as its argument, but `openDrawer` is a function that takes an optional boolean argument.

Can I keep it wrapped in the arrow function?